timescaledb/tutorials/nfl-fantasy-league.md

```diff
@@ -0,0 +1,198 @@
+    playDescription TEXT,
+    quarter INT,
+    down INT,
+    yardsToGo INT,
+    possessionTeam TEXT,
+    playType TEXT,
+    yardlineSide TEXT,
+    yardlineNumber INT,
+    offenseFormation TEXT,
+    personnelO TEXT,
+    defendersInTheBox INT,
+    numberOfPassRushers INT,
+    personnelD TEXT,
+    typeDropback TEXT,
+    preSnapVisitorScore INT,
+    preSnapHomeScore INT,
+    gameClock TIME,
+    absoluteYardlineNumber INT,
+    penaltyCodes TEXT,
+    penaltyJerseyNumber TEXT,
+    passResult TEXT,
+    offensePlayResult INT,
+    playResult INT,
+    epa DOUBLE PRECISION,
+    isDefensivePI BOOLEAN
+);
+
+CREATE TABLE tracking (
+    time TIMESTAMP,
+    x DOUBLE PRECISION,
+    y DOUBLE PRECISION,
+    s DOUBLE PRECISION,
+    a DOUBLE PRECISION,
+    dis DOUBLE PRECISION,
+    o DOUBLE PRECISION,
+    dir DOUBLE PRECISION,
+    event TEXT,
+    nflId INT,
+    displayName TEXT,
+    jerseyNumber INT,
+    position TEXT,
+    frameId INT,
+    team TEXT,
+    gameId INT,
+    playId INT,
+    playDirection VARCHAR(5),
+    route TEXT
+);
+```
+
+Add indexes to the `tracking` table speed up queries:
+
+```sql
+CREATE INDEX idx_gameid ON tracking (gameid);
+CREATE INDEX idx_playerid ON tracking (player_id);
+CREATE INDEX idx_playid ON tracking (playid);
+```
+
+## Ingest data from CSV files
+
+We have three separate CSV files for game, player, and play tables. For the tracking table, you will need to
+import data from 17 CSV files (1 file for each week in the season).
+
+Let's use a Python script that utilizes psycopg2's `copy_expert` function to ingest the data:
+
+```python
+import config
+import psycopg2
+
+# connect to the database
+conn = psycopg2.connect(database=config.DB_NAME, 
+                        host=config.HOST, 
+                        user=config.USER, 
+                        password=config.PASS, 
+                        port=config.PORT)
+
+# insert CSV file into given table
+def insert(csv_file, table_name):
+    cur = conn.cursor()
+    copy_sql = """
+            COPY {table} FROM stdin WITH CSV HEADER
+            DELIMITER as ','
+            """.format(table=table_name)
+    with open(csv_file, 'r') as f:
+        cur.copy_expert(sql=copy_sql, file=f)
+        conn.commit()
+        cur.close()
+
+print("Inserting games.csv")
+insert("data/games.csv", "game")
+
+print("Inserting plays.csv")
+insert("data/plays.csv", "play")
+
+print("Inserting players.csv")
+insert("data/players.csv", "player")
+
+# iterate over each week's CSV file and insert
+for i in range(1, 18): 
+    print("Inserting week{i}".format(str(i)))
+    insert("data/week{i}.csv".format(i=i), "tracking")
+
+conn.close()
+```
+
+## Analyze NFL data
+
+Now that you have all the data ingested, let's go over some ideas on how you can analyze the data to help you perfect
+your fantasy drafting strategy and win your fantasy season.
+
+To analyze the dataset you will need to install two libraries, if you haven't already: Pandas and Matplotlib.
```

Comment:
```suggestion
To analyze the dataset you need to install two libraries, if you haven't already: Pandas and Matplotlib.
```

Reply:
This sounds like a requirement, which it shouldn't be (generally). Feels like we should say something like, "For each example query below, we visualize the result with Matplotlib in Python. If you want to graph the data as well, you will need to import Pandas and Matplotlib."

Remember, we want to emphasize the SQL part of this, but so far I feel like I'm only reading a Python tutorial. To me, I think it needs to be clear up front that we're analyzing the data with SQL, and using Python as a way to interact with/visualize the results. IMO